src/main/java/com/google/sps/servlets/ConnectionRequestsServlet.java

```diff
@@ -100,19 +100,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     boolean success = false;
 
-    User user = dataAccess.getCurrentUser();
-    if (user != null) {
-      Mentee mentee = dataAccess.getMentee(user.getUserId());
-      if (mentee != null) {
-        MentorshipRequest mentorshipRequest =
-            dataAccess.getMentorshipRequest(Long.parseLong(requestKey));
-        if (mentorshipRequest != null) {
-          if (choice.equals(ACCEPT)) {
-            dataAccess.approveRequest(mentorshipRequest);
-            success = true;
-          } else if (choice.equals(DENY)) {
-            dataAccess.denyRequest(mentorshipRequest);
-            success = true;
+    Long requestDatastoreKey = null;
+    try {
+      requestDatastoreKey = Long.parseLong(requestKey);
+    } catch (NumberFormatException e) {
+    }
+    if (requestDatastoreKey != null && (choice.equals(ACCEPT) || choice.equals(DENY))) {
+      User user = dataAccess.getCurrentUser();
+      if (user != null) {
+        Mentee mentee = dataAccess.getMentee(user.getUserId());
+        if (mentee != null) {
+          MentorshipRequest mentorshipRequest =
+              dataAccess.getMentorshipRequest(requestDatastoreKey);
+          if (mentorshipRequest != null) {
+            if (choice.equals(ACCEPT)) {
+              dataAccess.approveRequest(mentorshipRequest);
+              success = true;
+            } else if (choice.equals(DENY)) {
+              dataAccess.denyRequest(mentorshipRequest);
+              success = true;
+            }
```

Comment:
this is error prone having such a deep nesting of ifs. Prefer to re-write this in a way that terminates the method when a condition is not met. For e.g.:

if (requestDatastoreKey == null || !(choice.equals(ACCEPT) || choice.equals(DENY))) {
return;
}
User user = ...
if (user == null) return;
Mentee mentee = ...
if (mentee == null) return;
...